velox/exec/SortBuffer.cpp

```diff
@@ -112,19 +114,29 @@ void SortBuffer::noMoreInput() {
     sortedRows_.resize(numInputRows_);
     RowContainerIterator iter;
     data_->listRows(&iter, numInputRows_, sortedRows_.data());
-    std::sort(
-        sortedRows_.begin(),
-        sortedRows_.end(),
-        [this](const char* leftRow, const char* rightRow) {
-          for (vector_size_t index = 0; index < sortCompareFlags_.size();
-               ++index) {
-            if (auto result = data_->compare(
-                    leftRow, rightRow, index, sortCompareFlags_[index])) {
-              return result < 0;
+    if (prefixSortConfig_.has_value() &&
```

Comment:
Why do we need this if-else branch? Can we not always use PrefixSort?

Reply:
·Waiting for all caller to pass prefixSortConfig

Reply:
The default prefixSortConfig needs to be configured based on query-config to maintain configuration consistency，however query-config is not passed to sort-buffer, we need to make prefixSortConfig one-by-one.

Reply:
Can we update all the callers to pass this config?

Reply:
I plan to  do this, but  maybe change a lot of code in one MR,  so I use optional to add them one-by-one, after all things could work , I will use 'const PrefixSort&&' instead of optional

Reply:
How many places need updating? Can you update them all and make sure everything works? We can then see how large the change is and decide whether it need splitting.

Reply:
Ok， I will try this after 5/06 (May Day International Labor Day), as soon as possible~